config/config-server/src/test/java/com/thoughtworks/go/config/MagicalGoConfigXmlLoaderTest.java

```diff
@@ -4171,6 +4175,45 @@ public void shouldMigrateDESEncryptedEnvironmentVariables_XslMigrationFrom108To1
         assertThat(new AESEncrypter(new AESCipherProvider(systemEnvironment)).decrypt(encryptedValue), is("user-password!"));
     }
 
+    @Test
+    public void shouldRemoveEmptySCMPasswordAndEncryptedPasswordAttributes_XslMigrationFrom109To110() throws Exception {
+        ReflectionUtil.setField(new DESCipherProvider(systemEnvironment), "cachedKey", null);
+        FileUtils.writeStringToFile(systemEnvironment.getDESCipherFile(), "269298bc31c44620", UTF_8);
+
+        String content = configWithPipeline(
+                "<pipeline name='some_pipeline'>"
+                        + "    <materials>"
+                        + "      <svn url='svn1' username='bob' encryptedPassword='' dest='svn1'/>"
+                        + "      <svn url='svn2' username='bob' password='' dest='svn2'/>"
+                        + "      <tfs url='tfsurl1' username='user' domain='domain' encryptedPassword='' projectPath='path' dest='tfs1' />"
+                        + "      <tfs url='tfsurl2' username='user' domain='domain' password='' projectPath='path' dest='tfs2' />"
+                        + "      <p4 port='host:9999' username='user' encryptedPassword='' dest='perforce1'>" +
+                        "          <view><![CDATA[view]]></view>" +
+                        "        </p4>"
+                        + "      <p4 port='host:9999' username='user' password='' dest='perforce2'>" +
+                        "          <view><![CDATA[view]]></view>" +
+                        "        </p4>"
+                        + "    </materials>"
+                        + "  <stage name='some_stage'>"
+                        + "    <jobs>"
+                        + "      <job name='some_job'>"
+                        + "      </job>"
+                        + "    </jobs>"
+                        + "  </stage>"
+                        + "</pipeline>", 109);
+
+        assertThat(XpathUtils.nodeExists(content, "//*[@password='']"), is(true));
+        assertThat(XpathUtils.nodeExists(content, "//*[@encryptedPassword='']"), is(true));
+
+
+        CruiseConfig config = ConfigMigrator.loadWithMigration(content).config;
+
+        String xmlPartial = xmlWriter.toXmlPartial(config);
+
+        assertThat(XpathUtils.nodeExists(xmlPartial, "//@password"), is(false));
+        assertThat(XpathUtils.nodeExists(xmlPartial, "//@encryptedPassword"), is(false));
```

Comment:
Also check if a non empty password or encryptedPassword exists, the nodeExists?

Reply:
There is no non-empty `password` or `encryptedPassword` in this test. This is covered in other tests, so I let it be.